### DISCUSSION TITLE
using the url in generateMetadata Function

### DISCUSSION THREAD

**Discussion by sagarchoudhary96** [vercel/next.js#49284] [Help] [answered]
### Summary

I need to create dynamic metadata based on the subdomain in my url so need to access the url in it, not sure how it can be done if it's supported at the moment.

### Additional information
Using Nextjs version -> 13.2.4

**Comment by sagarchoudhary96** [comment, marked as answer]
Found the solution, can read header using `import { headers } from "next/headers";` and read host from the header

**Reply by javad7z7** [reply]
Hey mate, can you give me an example of using this header in generateMetadata? This needs to be added to the RootLayout

**Reply by minlare** [reply]
https://github.com/vercel/next.js/discussions/50189#discussioncomment-6597167

**Reply by sahil-1012** [reply]
This doesn't work in next js 15, any solution for this anyone got yet?
